### PR TITLE
Add continuous testing for logs to k3d smoke test environment

### PR DESCRIPTION
**Pull request by tpaschalis**
#### PR Description
This PR adds Loki Canary to our K3D-based smoke test environment. It is meant as a test-drive to see how everything works together before adding it to a live environment.

#### Which issue(s) this PR fixes
No issue filed.

#### Notes to the Reviewer
I'm not sure whether the `write_relabel_configs` are needed, since the new scrape job will drop all pods other than the canary ones, I'll have to check.

I'm reusing the lib/loki jsonnet that we'd already have in our repo, it worked out nicely.

#### PR Checklist

- [ ] CHANGELOG updated
- [X] Documentation added
- [ ] Tests updated (N/A)

### CODE REVIEW

File: example/k3d/smoke/main.jsonnet [hunk 1]
```diff
@@ -86,6 +89,29 @@ local smoke = {
       configMap.mixin.metadata.withNamespace("smoke")
   },
 
+  loki: loki.new(namespace = 'smoke'),
+
+  // https://grafana.com/docs/loki/latest/operations/loki-canary/
+  canary : canary{
```
Comment (rfratto) [inline example/k3d/smoke/main.jsonnet:95]: You might want to install jsonnetfmt :)

```suggestion
  canary: canary{
```

Comment (tpaschalis) [reply example/k3d/smoke/main.jsonnet]: Thanks for the suggestion. Ran `:JsonnetFmt` and it added some linting fixes from previous PRs.

**Review by mattdurham — APPROVED** [review]
Love the diagrams.

**Review by rfratto — APPROVED** [review]
LGTM!

**Pull request merged**